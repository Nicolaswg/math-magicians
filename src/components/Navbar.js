import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Navbar.module.css';

function Navbar() {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: 3,
      path: '/qoute',
      text: 'Qoute',
    },
  ];
  return (
    <>
      <div className={styles.headerContainer}>
        <div className={styles.headerTitle}>
          <h1>Math Magicians</h1>
        </div>
        <nav className={styles.navbar}>
          <ul>
            {
        links.map((link) => (
          <>
            <li key={link.id}>
              <NavLink className={styles.activeLink} to={link.path}>
                {link.text}
              </NavLink>
            </li>
            <div className={styles.line} />
          </>
        ))
        }
          </ul>
        </nav>
      </div>
    </>
  );
}

export default Navbar;
