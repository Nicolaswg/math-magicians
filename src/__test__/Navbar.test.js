import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

it('Render Navbar correctly', () => {
  const tree = renderer.create(
    <Router>
      <Navbar />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
