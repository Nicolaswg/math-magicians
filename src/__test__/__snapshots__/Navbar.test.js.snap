// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Navbar correctly 1`] = `
<div
  className="headerContainer"
  data-testid="navbar"
>
  <div
    className="headerTitle"
  >
    <h1>
      Math Magicians
    </h1>
  </div>
  <nav
    className="navbar"
  >
    <ul>
      <li>
        <a
          aria-current="page"
          className="activeLink active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <div
        className="line"
      />
      <li>
        <a
          className="activeLink"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <div
        className="line"
      />
      <li>
        <a
          className="activeLink"
          href="/qoute"
          onClick={[Function]}
        >
          Qoute
        </a>
      </li>
      <div
        className="line"
      />
    </ul>
  </nav>
</div>
`;
